python/lsst/ts/cbp/component.py

```diff
@@ -143,14 +122,33 @@ def parse_reply(self):
             The reply that was parsed.
 
         """
-        if self.simulation_mode == 0:
-            parsed_reply = self.socket.recv(128).decode('ascii').split("\r")[0]
-        elif self.simulation_mode == 1:
-            parsed_reply = "0"
-        return parsed_reply
+        async with self.lock:
```

Comment:
I'm afraid this lock does not have the desired effect. The intent of the lock is to prevent a race condition where:
- task A issues a command
- task A is interrupted by task B before A can read replies
- task B issues a command and reads replies
- task B then sees the reply for task A's command, which is bad, and if B reads all replies then task A sees no replies when it finally gets control again

Thus the correct sequence is to acquire the lock, issue the command, read the replies for that command, then release the lock. In other words:
```
async with self.lock:
    write command
    read replies
```

I _very strongly_ suggest combining "send a command" and "read replies" into a single method and have that method acquire the lock. As long as "send a command" and "read replies" are separate methods it is nearly impossible to safely lock access to the socket.